plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/_impl.py

```diff
@@ -158,14 +161,32 @@ def sweep(self, arguments: List[str]) -> None:
             )
             self.validate_batch_is_legal(overrides)
             returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
-            self.job_idx += len(returns)
             # would have been nice to avoid waiting for all jobs to finish
             # aka batch size Vs steady state (launching a new job whenever one is done)
+            self.job_idx += len(returns)
+            # check job status and prepare losses
+            failures = 0
             for cand, ret in zip(candidates, returns):
-                loss = direction * ret.return_value
-                optimizer.tell(cand, loss)
-                if loss < best[0]:
-                    best = (loss, cand)
+                if ret.status == utils.JobStatus.COMPLETED:
+                    rectified_loss = direction * ret.return_value
+                else:
+                    rectified_loss = math.inf
+                    failures += 1
+                    try:
+                        ret.return_value
+                    except HydraJobException as e:
+                        log.warning(f"Returning infinity for failed experiment: {e}")
```

Comment:
Is there a scenario where this does not raise?
If yes, what is it?
if no, can you remove the try-except and just print the warning?

Reply:
if I am correct this always raises since the job is not COMPLETED. My aim here is to capture the error so that I can log it: I tend to print as much information as I can so that it's easier to debug, being robust to the failures does not mean they are expected.
Does this makes sense to you?

Reply:
It does. I wonder if we should add another property to access the exception that will raise if there is no exception.

Reply:
In concurrent.futures, jobs have an `exception()` method that returns the exception if it exists, and `None` otherwise. This way users can handle it the way they want.